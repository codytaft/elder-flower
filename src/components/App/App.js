import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';

import './App.css';

class App extends Component {
  render() {
    return (
      <div className="app">
        <header className="app-header">
          <h1 className="app-title">Welcome to Elder Flower</h1>
        </header>
      </div>
    );
  }
}

export default App;
