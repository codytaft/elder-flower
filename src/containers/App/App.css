.app {
  font-family: 'Nunito', sans-serif;
  height: 100vh;
  overflow: scroll;
  text-align: center;
  background-image: linear-gradient(to top, #eacda3, #c7c293, #a3b78b, #80aa88, #5f9c89); }

.app-hidden {
  height: 100%; }

.app-header {
  height: 150px;
  padding: 20px;
  color: white;
  font-size: 32px; }
