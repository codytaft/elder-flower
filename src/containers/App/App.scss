.app {
  font-family: 'Nunito', sans-serif;
  height: 100vh;
  text-align: center;
  background-image: linear-gradient(
    to top,
    #eacda3,
    #c7c293,
    #a3b78b,
    #80aa88,
    #5f9c89
  );
}

.app-header {
  height: 150px;
  padding: 20px;
  color: white;
  font-size: 32px;
}
