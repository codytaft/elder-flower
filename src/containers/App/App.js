import React, { Component } from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import Routes from '../../components/Routes/Routes';
import { Route, NavLink } from 'react-router-dom';

import './App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="app">
          <header className="app-header">
            <h1 className="app-title">Welcome to Elder Flower</h1>
          </header>
          <main className="main-container">
            <NavLink exact to="/login" className="nav-link nav-link-login">
              Login
            </NavLink>
            <NavLink exact to="/sign-up" className="nav-link nav-link-signup">
              Sign Up
            </NavLink>
            <Routes />
          </main>
        </div>
      </Router>
    );
  }
}

export default App;
