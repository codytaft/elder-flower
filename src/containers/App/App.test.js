import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should match the snapshot', () => {
    let wrapper;
    wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if location.pathname equals sign-up-elder', () => {
    let wrapper;
    const mockLocation = { pathname: '/sign-up-elder' };
    wrapper = shallow(<App location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot if location.pathname does not equal sign-up-elder', () => {
    let wrapper;
    const mockLocation = { pathname: '/' };
    wrapper = shallow(<App location={mockLocation} />);
    expect(wrapper).toMatchSnapshot();
  });
});
