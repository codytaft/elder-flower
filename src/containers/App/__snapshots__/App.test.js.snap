// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(App) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`App should match the snapshot if location.pathname does not equal sign-up-elder 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(App)
    location={
        Object {
            "pathname": "/",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`App should match the snapshot if location.pathname equals sign-up-elder 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(App)
    location={
        Object {
            "pathname": "/sign-up-elder",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
